### PR TITLE
Groups FE Trends filtering 

### PR DESCRIPTION
**Pull request by macobo**
- Add /api/projects/@current/groups/property_definitions
- Add routes for fetching group types
- FE: Group analytics feature flag
- FE: Group analytics base types for properties
- FE: Initial implementation of fetching logics

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: frontend/src/lib/components/TaxonomicFilter/types.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export interface TaxonomicFilterLogicProps extends TaxonomicFilterProps {
 
 export interface TaxonomicFilterGroup {
     name: string
-    type: TaxonomicFilterGroupType
+    type: TaxonomicFilterGroupType | string
```
Comment (macobo) [inline frontend/src/lib/components/TaxonomicFilter/types.ts:27]: so this kind of type change is problematic because we're losing most validation here. What's the problem you're running into here? Can it be solved by modifying `TaxonomicFilterGroupType`?

File: frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.ts [hunk 2]
```diff
@@ -70,8 +72,17 @@ export const taxonomicFilterLogic = kea<taxonomicFilterLogicType>({
             (taxonomicFilterLogicKey) => taxonomicFilterLogicKey,
         ],
         taxonomicGroups: [
-            (selectors) => [selectors.currentTeamId],
-            (teamId): TaxonomicFilterGroup[] => [
+            (selectors) => [selectors.currentTeamId, selectors.groupTypes],
+            (teamId, groupTypes): TaxonomicFilterGroup[] => [
+                ...(groupTypes.length > 0
```
Comment (macobo) [inline frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.ts:77]: This belongs to a separate selector - putting it here just makes this selector really hard to follow, give it it's own name.

Also, worth checking the feature flag here as well?

File: frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.ts [hunk 3]
```diff
@@ -70,8 +72,17 @@ export const taxonomicFilterLogic = kea<taxonomicFilterLogicType>({
...
+                    ? groupTypes.map((gt) => ({
+                          name: capitalizeFirstLetter(gt.group_type),
+                          type: gt.group_type,
+                          endpoint: `api/projects/${teamId}/groups`,
+                          getName: (group: GroupType) => group.group_type,
+                          getValue: (group: GroupType) => group.group_type_index,
+                      }))
+                    : []),
```
Comment (macobo) [inline frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.ts:85]: The ternary is not needed - if you have an empty array then mapping over an empty array yields an empty array.

File: frontend/src/lib/components/PropertyFilters/components/TaxonomicPropertyFilter.tsx [hunk 4]
```diff
@@ -53,6 +54,7 @@ export function TaxonomicPropertyFilter({
             }}
             taxonomicGroupTypes={
                 taxonomicGroupTypes || [
+                    ...(groupTypes.length > 0 ? [...groupTypes.map((gt) => gt.group_type)] : []),
```
Comment (macobo) [inline frontend/src/lib/components/PropertyFilters/components/TaxonomicPropertyFilter.tsx:57]: Couple of issues:
1. This business logic does not belong on the component.
2. Wouldn't this enable groups everywhere groups aren't explicitly enabled? This will cause bugs since we won't support groups from the get-go everywhere.

We need a pattern to enable groups (ideally by adding one line to props in callsites) component-by-component.

File: frontend/src/models/groupsModel.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+export const groupsModel = kea<groupsModelType>({
+    connect: {
+        values: [teamLogic, ['currentTeamId']],
+    },
+    loaders: ({ values }) => ({
+        groupTypes: [
+            [] as Array<GroupType>,
+            {
+                loadAllGroupTypes: async () => {
+                    if (featureFlagLogic.values.featureFlags[FEATURE_FLAGS.GROUP_ANALYTICS]) {
```
Comment (macobo) [inline frontend/src/models/groupsModel.ts:18]: So we should also check whether clickhouse is enabled here.

Suggestion: Extract a selector for the feature flag check + checking whether clickhouse is enabled, use it elsewhere around here.

**Comment by macobo** [comment]
What does the modal look like with these changes? Please include screenshots to make giving feedback easier :)

File: frontend/src/lib/components/TaxonomicFilter/types.ts (hunk 1 above)
Comment (liyiy) [reply frontend/src/lib/components/TaxonomicFilter/types.ts:27]: The biggest problem I'm running into here is that because the group types are dynamic, I'm not sure how to set that in the typing besides leaving it as a `string` option

Comment (macobo) [reply frontend/src/lib/components/TaxonomicFilter/types.ts:27]: Since we have limited amount of groups (5) we could hard-code them into the GroupType.

File: frontend/src/lib/components/PropertyFilters/components/TaxonomicPropertyFilter.tsx (hunk 4 above)
Comment (liyiy) [reply frontend/src/lib/components/PropertyFilters/components/TaxonomicPropertyFilter.tsx:57]: Hmm I think for point 2 it shouldn't enable groups because that's feature flagged in the `groupsModel` so when `groupTypes` returns an empty array, nothing shows up, though I can protect this more

File: frontend/src/lib/components/TaxonomicFilter/TaxonomicFilter.tsx [hunk 6]
```diff
@@ -51,6 +51,7 @@ export function TaxonomicFilter({
     }, [])
 
     const placeholder = taxonomicGroupTypes
+        .filter(type => !type.includes('groups'))
```
Comment (liyiy) [inline frontend/src/lib/components/TaxonomicFilter/TaxonomicFilter.tsx:54]: filtering this out for now since it's appending weird names like `groups_0`, `groups_1`

**Comment by liyiy** [comment]
Recording of the group tabs and populated properties - 

Setting the property itself doesn't work yet - it's been tricky to find out how everything is connected/calling each other, so opting to get this merged in before adding that so we can iterate faster?

Also the call for groups property values at `api/groups/values/key=whatever` doesn't work yet, so I need to also add that I think (or can this use the `api/person/values/key=whatever`??)

https://user-images.githubusercontent.com/25164963/140471311-ee0f30f1-cf88-4c29-acc9-74dcaca2c28a.mov

**Comment by macobo** [comment]
Feel free to merge once this is green :)
